pkg/util/vmopv1/resize_overwrite.go

```diff
@@ -85,6 +89,44 @@ func overrideMMIOSize(
 	return out
 }
 
+func clearMMPowerOffEC(
+	_ vmopv1.VirtualMachine,
+	ci vimtypes.VirtualMachineConfigInfo,
+	_ *vimtypes.VirtualMachineConfigSpec) []vimtypes.BaseOptionValue {
+
+	// Ensure MMPowerOffVMExtraConfigKey is no longer part of ExtraConfig as
+	// setting it to an empty value removes it.
+
+	v, ok := util.OptionValues(ci.ExtraConfig).GetString(constants.MMPowerOffVMExtraConfigKey)
+	if !ok || v == "" {
+		return nil
+	}
+
+	return []vimtypes.BaseOptionValue{
+		&vimtypes.OptionValue{Key: constants.MMPowerOffVMExtraConfigKey, Value: ""},
+	}
+}
+
+func updateV1Alpha1CompatibleEC(
+	vm vmopv1.VirtualMachine,
+	ci vimtypes.VirtualMachineConfigInfo,
+	_ *vimtypes.VirtualMachineConfigSpec) []vimtypes.BaseOptionValue {
+
+	bs := vm.Spec.Bootstrap
```

Comment:
Nit: Add a comment about why this check is needed. vAppConfig + LinuxPrep ==  OvfEnv ?.